### ISSUE TITLE
getting smoother stroke AA

### ISSUE THREAD

**Issue by wtholliday** [memononen/nanovg#215]
How can I get smoother AA for strokes? Here's a comparison of nanovg:

![nanovg](https://cloud.githubusercontent.com/assets/841144/6952884/8f240c34-d87a-11e4-94e7-efb8294cec9c.png)

vs. my old code:

![mycode](https://cloud.githubusercontent.com/assets/841144/6952892/968768f4-d87a-11e4-943f-a484c404797c.png)

My old code uses the smoothstep function for AA falloffs. Nanovg seems to use a box filter, no?

(See section 3.17.3 of Compute Graphics: Principles and Practice for some good discussion about this stuff)

**Comment by wtholliday** [comment]
I doubled the `fringeWidth`, which helps a lot.

I also tried changing `strokeMask` to use `smoothstep`, but this seemed like not much of a difference. Perhaps the smoothstep results in a slightly more defined edge because it is actually a more sudden transition than the existing linear one, but it's very hard to tell.

**Comment by memononen** [comment]
Yes, the AA is equivalent to box filter. There are number of things that affect the AA quality, one of them that is a bit hard to control is gamma correction. It can have a huge difference.

Just widening the width generally makes things blurry. Widening the width plus adding smoothstep should have a bit more gaussian like result. Doubling the fringe plus using Perlin's smootherstep could be a good solution.

**Comment by wtholliday** [comment]
Yeah, I ended up seeing a little blurryness in certain spots (perhaps due to the colors involved), and went for smoothstep, which looks good! Since gamma can't be controlled, why not use smoothstep and a wider fringeWidth by default?

**Comment by memononen** [comment]
I just wanted to keep things simple.

**Comment by wtholliday** [comment]
It was a simple change, assuming I did it right :)

https://github.com/wtholliday/nanovg/commit/27ac04132934388abc0be7d2008e25d5fb9d4405
